### PR TITLE
Add-Deepseek

### PR DESCRIPTION
**Pull request by Inlei**
# Description

New Model Provider - Deepseek

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/core/model_runtime/model_providers/deepseek/llm/deepseek-chat.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    default: 0.5
+  - name: top_p
+    use_template: top_p
+    min: 0
+    max: 1
+    default: 1
+  - name: max_tokens
+    use_template: max_tokens
+    min: 1
+    max: 1024
```
Comment (takatost) [inline api/core/model_runtime/model_providers/deepseek/llm/deepseek-chat.yaml]: maybe 32000?

File: api/core/model_runtime/model_providers/deepseek/llm/deepseek-coder.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    default: 0.5
+  - name: top_p
+    use_template: top_p
+    min: 0
+    max: 1
+    default: 1
+  - name: max_tokens
+    use_template: max_tokens
+    min: 1
+    max: 1024
```
Comment (takatost) [inline api/core/model_runtime/model_providers/deepseek/llm/deepseek-coder.yaml]: the same issue

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/deepseek/llm/deepseek-coder.yaml (hunk 2 above)
Comment (Inlei) [reply api/core/model_runtime/model_providers/deepseek/llm/deepseek-coder.yaml]: ok

File: api/core/model_runtime/model_providers/deepseek/llm/deepseek-chat.yaml (hunk 1 above)
Comment (Inlei) [reply api/core/model_runtime/model_providers/deepseek/llm/deepseek-chat.yaml]: ok

**Review by takatost — APPROVED** [review]

**Pull request merged**